gu-hub-api.yaml

```diff
@@ -0,0 +1,582 @@
+      nodeId:
+        type: string
+        description: node public key hash in etherium format
+        example: '0x875f272d3b9e7b55a5784a131a60bf3d7a42c73c'
+        readOnly: true
+      tags:
+        type: array
+        items:
+          type: string
+  HubSession:
+    properties:
+      id:
+        type: string
+        description: Uniqe HUB Session identifier
+        example: 47f1dc88-0282-4ac3-f765-02b303e8c278
+        readOnly: true
+      created:
+        type: string
+        description: Session creation timestamp
+        format: date-time
+        readOnly: true
+      expires:
+        type: string
+        format: date-time
+      allocation:
+        type: string
+        description: resource allocation mode
+        enum:
+          - manual
+          - auto
+      name:
+        type: string
+        description: optional human readable name
+      tags:
+        type: array
+        uniqueItems: true
+        items:
+          type: string
+          pattern: '^[a-zA-Z][a-zA-Z0-9_:-]*$'
+  PeerSessionSpec:
+    properties:
+      envType:
+        $ref: '#/definitions/EnvType'
+      image:
+        properties:
+          hash:
+            type: string
+            description: Consistency hash
+          url:
+            type: string
+            description: Image location spec
+      name:
+        type: string
+        description: human readable name
+      tags:
+        type: array
+        description: peer session tags
+        uniqueItems: true
+        items:
+          type: string
+  Command:
+    type: object
+    properties:
+      exec:
+        $ref: '#/definitions/ExecCommand'
+      start:
+        $ref: '#/definitions/StartCommand'
+      stop:
+        $ref: '#/definitions/StopCommand'
+      addTags:
+        type: array
+        uniqueItems: true
+        items:
+          type: string
+      delTags:
+        type: array
+        uniqueItems: true
+        items:
+          type: string
+      downloadFile:
+        $ref: '#/definitions/DownloadFileCommand'
+      uploadFile:
+        $ref: '#/definitions/UploadFileCommand'
+  ExecCommand:
+    description: synchronous exec of session entry point
+    type: object
+    properties:
+      executable:
+        type: string
+      args:
+        type: array
+        items:
+          type: string
+  StartCommand:
+    properties:
+      executable:
+        type: string
+      args:
+        type: array
+        items:
+          type: string
+  StopCommand:
+    properties:
+      childId:
+        description: 'id of asynchronous process, started with StartCommand'
+        type: string
+  DownloadFileCommand:
+    properties:
+      uri:
+        type: string
+        format: url
+      filePath:
+        type: string
+  UploadFileCommand:
+    properties:
+      uri:
+        type: string
+        format: url
+      filePath:
+        type: string
+  ConfigStash:
+    description: 'Free style configuration object'
```

Comment:
```suggestion
    description: Free style configuration object
```

Reply:
I do not understand. This description is for a type not for operation.

Reply:
this is controversial, but adding all possible Commands to operation with their respective return values would result in the long description there, which would easily lose sync with commands when we change existing, or add a new one. So I propose to describe commands here